plugins/techdocs/src/alpha.tsx

```diff
@@ -157,10 +157,13 @@ const techDocsReaderPage = PageBlueprint.make({
  */
 const techDocsEntityContent = EntityContentBlueprint.makeWithOverrides({
   inputs: {
-    emptyState: createExtensionInput([coreExtensionData.reactElement], {
-      singleton: true,
-      optional: true,
-    }),
+    emptyState: createExtensionInput(
+      [coreExtensionData.reactElement.optional()],
```

Comment:
This does not seem right to me. Why would you be able to optionally supply an empty content renderer, that optionally returns react data? Just don't supply an empty content renderer instead. And then the consumer of this input just renders null or some default, if the input wasn't populated.

I think the root problem is that the `techDocsEntityContentEmptyState` exists at all below. Maybe it should be removed?

If it is kept, maybe make it have a non-optional output, and return null there?

Reply:
Sorry for the slow reply here! We've talked about this a bit offline but I'm following up here for posterity. 

I think you raise a good point. The reason the `techDocsEntityContentEmptyState` extension exists is to give adopters overriding it something typed that they can use:

```typescript
const customEmptyState = techdocsPlugin
  .getExtension('empty-state:techdocs/entity-content')
  .override({
    factory: () => 
      [
        coreExtensionData.reactElement(
          <div>Hello empty state!</div>
        )
      ]
  })
```

Without this, the adopter has to write something like this, without any help from the type system:

```typescript
const customEmptyState = createExtension({
  kind: 'empty-state',
  name: 'entity-content',
  attachTo: { 
    id: 'entity-content:techdocs', 
    input: 'emptyState' 
  },
  output: [coreExtensionData.reactElement],
  factory: () => 
    [
      coreExtensionData.reactElement(
        <div>Hello empty state!</div>
      )
    ]
});
```

With this in mind, I wonder if we should consider a more general-purpose solution for providing extension inputs? That might avoid the incentive to build helper extensions in this situation while still preserving the egonomics for adopters. Perhaps an API something like:

```typescript
// Returns an extension which attaches to 
// the entity content extension and supplies 
// the 'emptyState' input. Types of `input` 
// and `factory` are inferred from the type
// of the extension, so you get suggestions
// when writing the code and a guarantee that
// the resulting extension will be compatible.
const customEmptyState = techdocsPlugin
  .getExtension('empty-state:techdocs/entity-content')
  .makeInputExtension({
    input: 'emptyState',
    factory: () => 
      [
        coreExtensionData.reactElement(
          <div>Hello empty state!</div>
        )
      ]
  })
```